.github/workflows/main.yml

```diff
@@ -0,0 +1,38 @@
+name: main
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - "*"
+  workflow_dispatch: {}
+
+jobs:
+  test:
+    name: Elixir ${{ matrix.elixir }} OTP ${{ matrix.otp }}
+    strategy:
+      matrix:
+        otp: ['27', '26', '25']
+        elixir: ['1.18.3', '1.17.3', '1.16.3']
+        exclude:
+          # Unsupported
+          - elixir: '1.16.3'
+            otp: '27'
+    runs-on: 'ubuntu-24.04'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{ matrix.otp }}
+          elixir-version: ${{ matrix.elixir }}
+      - run: mix deps.get
+      - run: mix compile
+      - run: mix format --check-formatted
+      - run: mix coveralls
+        if: ${{ (matrix.otp == '27') && (matrix.elixir == '1.18.3' ) }}
+      - run: mix docs
+        if: ${{ (matrix.otp == '27') && (matrix.elixir == '1.18.3' ) }}
+      - run: mix dialyzer
+        if: ${{ (matrix.otp == '27') && (matrix.elixir == '1.18.3' ) }}
```

Comment:
Just one note, don't we want to keep running `Credo` here in CI as well?

Reply:
yes, I forgot to add it